### PR TITLE
[doc] Update external.md

### PR DESCRIPTION
**Pull request by vai9er**
Issue: #

### CODE REVIEW

**Review by neozhaoliang — APPROVED** [review]
Thanks, and LGTM!

File: docs/lang/articles/basic/external.md [hunk 1]
```diff
@@ -11,15 +11,15 @@ We use NumPy arrays as an example to illustrate the data transfer process becaus
...
+2. Pass `arr` as an argument to a kernel or a Taichi function using `ti.types.ndarray()` as type hint. The argument is passed by reference without creating a copy of `arr`. Thus, any modification to this argument from inside a kernel or Taichi function also changes the original array `arr`. This approach is preferred when the kernel or Taichi function that takes in the argument needs to process the original array (for storage or filtering, for example).
 
 :::note
 `from_numpy() / from_torch()` can take in any numpy array or torch Tensor, no matter it's contiguous or not. Taichi will manage its own copy of data. However, when passing an argument to a Taichi kernel, only contiguous numpy arrays or torch Tensors are supported.
 :::
 
-## Data transfer between NumPy arrays and Taichi fields
+## Data Transfer between NumPy arrays and Taichi Fields
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:22]: We use the title case for document titles, and the sentence case for subheadings. So only the major words in first-level headings are capitalized,

File: docs/lang/articles/basic/external.md [hunk 2]
```diff
@@ -46,22 +46,25 @@ arr = x.to_numpy()
 #       [6, 7, 8]], dtype=int32)
 ```
 
-## Data transfer between PyTorch/Paddle tensors and Taichi fields
+## Data Transfer between PyTorch/Paddle Tensors and Taichi Fields
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:49]: ibid.

File: docs/lang/articles/basic/external.md [hunk 3]
```diff
@@ -132,7 +135,7 @@ As mentioned before, when transferring data between a `ti.field/ti.Vector.field/
     field.from_numpy(array)  # the input array must be of shape (256, 512, 3, 4)
     ```
 
-- When importing data to a struct field, export the data of the corresponding external array as **a dictionary of NumPy arrays, PyTorch tensors, or Paddle tensors** with keys being struct member names and values being struct member arrays. Nested structs are exported as nested dictionaries:
+4. When importing data to a struct field, export the data of the corresponding external array as **a dictionary of NumPy arrays, PyTorch tensors, or Paddle tensors** with keys being struct member names and values being struct member arrays. Nested structs are exported as nested dictionaries:
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:138]: I think we should still use the bullet list here. We can discuss it in our next meeting.

File: docs/lang/articles/basic/external.md [hunk 4]
```diff
@@ -194,9 +199,11 @@ def test(a: ti.types.ndarray(), b: ti.types.ndarray()):  # assume a, b have the
             b[i, j] = a[i, j] - (a[i-1, j] + a[i, j-1] + a[i+1, j] + a[i, j+1]) / 4
 ```
 
-This code snippet is more readable than the NumPy version above and runs way faster even on the CPU backend.
+Not only is this code snippet is more readable than the NumPy version above, but it also runs way faster even on the CPU backend.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:202]:
```suggestion
Not only is this code snippet more readable than the NumPy version above, but it also runs way faster even on the CPU backend.
```

File: docs/lang/articles/basic/external.md [hunk 5]
```diff
@@ -169,13 +172,15 @@ print(a)
 
 This is an entry-level example to show you how to call `ti.types.ndarray()`. We now illustrate a more advanced usage of this method.
 
-Assume that `a` and `b` are both 2D arrays of the same shape and dtype. For each cell `(i, j)` in `a`, we want to calculate the difference between its value and the average of its four neighboring cells while storing the result in the corresponding cell in `b`. In this case, cells on the boundary, which are cells with fewer than four neighbors, are ruled out for simplicity. This operation is usually denoted as the *discrete Laplace operator*:
+### Advanced
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:175]: Should we avoid a single third-level heading in one section? We can discuss it.

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

File: docs/lang/articles/basic/external.md [hunk 6]
```diff
@@ -11,15 +11,15 @@ We use NumPy arrays as an example to illustrate the data transfer process becaus
 
 There are two ways to import a NumPy array `arr` to the Taichi scope:
 
-- Create a Taichi field `f`, whose shape and dtype match the shape and dtype of `arr`, and call `f.from_numpy(arr)` to copy the data in `arr` into `f`. This approach is preferred when the original array is visited frequently from elsewhere in the Taichi scope (for example, in the case of texture sampling).
+1. Create a Taichi field `f`, whose shape and dtype match the shape and dtype of `arr`, and call `f.from_numpy(arr)` to copy the data in `arr` into `f`. This approach is preferred when the original array is visited frequently from elsewhere in the Taichi scope (for example, in texture sampling).
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:14]:
```suggestion
- Create a Taichi field `f`, whose shape and dtype match the shape and dtype of `arr`, and call `f.from_numpy(arr)` to copy the data in `arr` into `f`. This approach is preferred when the original array is visited frequently from elsewhere in the Taichi scope (for example, in texture sampling).
```

File: docs/lang/articles/basic/external.md (hunk 1 above)
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:22]:
```suggestion
## Data transfer between NumPy arrays and Taichi fields
```

File: docs/lang/articles/basic/external.md (hunk 2 above)
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:49]:
```suggestion
## Data transfer between PyTorch/Paddle tensors and Taichi fields
```

File: docs/lang/articles/basic/external.md [hunk 7]
```diff
@@ -146,7 +149,7 @@ As mentioned before, when transferring data between a `ti.field/ti.Vector.field/
     field.from_numpy(array_dict) # the input array must have the same keys as the field
     ```
 
-## Using external arrays as Taichi kernel arguments
+## Using External Arrays as Taichi Kernel Arguments
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:152]:
```suggestion
## Using external arrays as Taichi kernel arguments
```

File: docs/lang/articles/basic/external.md [hunk 8]
```diff
@@ -146,7 +149,7 @@ As mentioned before, when transferring data between a `ti.field/ti.Vector.field/
...
 
 Use type hint `ti.types.ndarray()` to pass external arrays as kernel arguments. For example:
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:154]:
```suggestion
Use type hint `ti.types.ndarray()` to pass external arrays as kernel arguments.

### An entry-level example

The following example shows the most basic way to call `ti.types.ndarray()`:
```

File: docs/lang/articles/basic/external.md [hunk 9]
```diff
@@ -169,13 +172,15 @@ print(a)
 
 This is an entry-level example to show you how to call `ti.types.ndarray()`. We now illustrate a more advanced usage of this method.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:173]:
```suggestion
### Advanced usage
```

File: docs/lang/articles/basic/external.md (hunk 5 above)
Comment (Olinaaaloompa) [inline docs/lang/articles/basic/external.md:175]:
```suggestion

```

**Review by Olinaaaloompa — CHANGES_REQUESTED** [review]

**Review by Olinaaaloompa — APPROVED** [review]
LGTM!

**Review by neozhaoliang — APPROVED** [review]
Thanks!

**Pull request merged**